### ISSUE TITLE
[Bug] Compilation fails when supplying --compile/--output

### ISSUE THREAD

**Issue by naltun** [chaos-lang/chaos#80]
**Describe the bug**
When **--compile** or **--output** are used to compile, there are various errors.

**To Reproduce**
1. Go to your Chaoslang spell directory
2. Run the following:
```shell
$ chaos -c fizzbuzz.kaos --output fizzbuzz
$ chaos --compile fizzbuzz.kaos -o fizzbuzz
$ chaos --compile fizzbuzz.kaos --output fizzbuzz
```

Each has its own error.

**Expected behavior**
Compilation should proceed successfully.

**Screenshots**
<img width="780" alt="image" src="https://user-images.githubusercontent.com/7507990/101169517-10d70800-360b-11eb-982c-1115a248a52d.png">

**Environment:**
 - CPU Architecture: **x86_64**
 - Operating System - Distro : **Arch Linux**
 - C Compiler Name and Version: **gcc - 10.2.0**
 - Chaos Language Version [e.g. `Chaos Language 0.1.0 (Nov 30 2020 04:39:36)`]

**Comment by mertyildiran** [comment]
@naltun I forgot to do a last minute check on command-line arguments especially the long options. So I've published `0.1.0` with this major bug which is too bad. We will have to create a patch release just because of this bug.

I've fixed this bug with the commits below and added tests that checking if there are any issues regarding the command-line arguments so that this issue will never happen again hopefully:

- https://github.com/chaos-lang/chaos/commit/292d4853f0c14fabe98479ec4dff9841bbaaaba6
- https://github.com/chaos-lang/chaos/commit/cdda07d028a9406ce9f7f338a16642fd04490186
- https://github.com/chaos-lang/chaos/commit/07fc33598a1fb84f5133ba24a29cdc20bf4684c9
